### DISCUSSION TITLE
Wifi activation on rpi w2

### DISCUSSION THREAD

**Discussion by debonaire007** [rern/rAudio#1824] [General]
How does one turn on Wifi on the rpi w2? Using rpi imager and setting it up there before burning the image does not seem to work. Raudio seems to boot, but is not seen to the network ...

**Comment by R101G** [comment]
I have never had much luck with creating the wifi file in BOOT. It always seems to end up in AP mode. The surest way after burning the image is to create a file in /etc/netctl named as the SSID and with the same content as you would put in the wifi0 file.

**Comment by debonaire007** [comment]
Thanks for the response, it is appreciated. I will look into it ... By the way, I confirm that creating 'wpa_supplicant.conf'  in the Boot directory does not work.

**Reply by R101G** [reply]
There is a wifi0 file in BOOT. If you open it you will see instructions. I thought that was what you were talking about, but if you haven't already tried it, it might work for you.

**Reply by kan-ibal** [reply]
All you need is `wifi0` file in BOOT
with YOUR wifi network NAME an PASSWORD
```
# Replace NAME and PASSWORD
# Save this file as 'wifi'

Interface=wlan0
Connection=wireless
IP=dhcp
Security=wpa
ESSID="NAME"
Key="PASSWORD"

```

**Reply by ceejayemm** [reply]
I always use the method, for all my RPi, mentioned by @kan-ibal above.  I find it best if the required password is NOT surround by double (or any) quotes eg Key=password

Chris

**Reply by debonaire007** [reply]
Regrettably, the sensible method suggested by Kan-Ibal does not work for my pi zero W. I saved the file as wifi on one try, and as wifi0 on another, with no luck. My ssid and pswd are correct. It should work, but does not, what may be wrong?

**Reply by R101G** [reply]
@debonaire007 
Try the alternative method of creating a file the same as your /boot/wifi, but with a filename of yourSSID instead of wifi, in /etc/netctl/. If it already exists, you might have to wait for another bug fix.

**Reply by debonaire007** [reply]
Will do thanks!

**Comment by debonaire007** [comment]
Super Kan-Ibal, I will try your suggestion. Know however that I pretty much did this with putting in the 'wpa_supplicant.conf' file. However it may be that this is a Debian 'thing´ and Raudio is Arch, we'll see ...

**Comment by debonaire007** [comment]
One more question, Raudio is A++, as we all the users know. However, is there an android app one can use from oneś cell phone? What is in the android app store 'says' not suitable for your device, or something like that. The Rune audio app was really good ...

**Reply by R101G** [reply]
APK is here: https://cloud.s-t-franz.de/s/SG82dQHRr27sjFS

**Reply by kan-ibal** [reply]
You can use RuneAudio app for rAudio as well. It is a fullscreen web browser that you can use for any player, just like Moode or Volumio.

**Reply by debonaire007** [reply]
Thanks for the info. What is the APK suggested by R101G? How do you install that on an android phone? In any case, I am pleased to hear that the old RuneAudio app works., but in my hands no luck. I guess by default it looks for an RuneAudio.local device and I have not figured ou how to change that, will try again ...

On a different note, this Forum is very civil and friendly. A breath of fresh air, compared to the usual user forums ... THANKS  ALL :)

**Reply by R101G** [reply]
@debonaire007 
Just do a search for how to install an APK. Basically, download it to your device, and click on it in the download folder. You might have to allow some permissions to the file manager or browser, depending on which one installs it.

**Reply by kan-ibal** [reply]
> Thanks for the info. What is the APK suggested by R101G? How do you install that on an android phone? In any case, I am pleased to hear that the old RuneAudio app works., but in my hands no luck. I guess by default it looks for an RuneAudio.local device and I have not figured ou how to change that, will try again ...

I do not have  RuneAudio.local but rAudio.local configured in /etc/avahi/services/rAudio.service
```
<?xml version="1.0" standalone='no'?>
<!DOCTYPE service-group SYSTEM "avahi-service.dtd">
<service-group>
<name replace-wildcards="yes">rAudio [%h] [XX:XX:XX:XX:XX:XX]</name>
  <service>
    <type>_http._tcp</type>
    <subtype>_app_._sub_._http._tcp</subtype>
    <port>80</port>
  </service>
</service-group>
```

where XX:XX:XX:XX:XX:XX is mac address.

**Reply by debonaire007** [reply]
> @debonaire007 Just do a search for how to install an APK. Basically, download it to your device, and click on it in the download folder. You might have to allow some permissions to the file manager or browser, depending on which one installs it.

I can confirm that the APK works very well and is easy to install as instructed, thank you.

**Reply by debonaire007** [reply]
Regretably, the rAudio APK while easy to install does not seem to have full functionality the way RuneAudio does. All I see on my cell phone is what song is playing and the ability to pause the song. There are no other controls and there is no way I found  to select what song to play, etc. What may be wrong?

**Reply by kan-ibal** [reply]
rAudio  should look like RuneAudio. Exactly the same, there is no visible difference. What is it look like in your Android default web Browser?

**Reply by debonaire007** [reply]
![image](https://github.com/user-attachments/assets/3f0e75a4-daa6-43dc-80eb-89fa0d55dbaa)
Here it what it looks on my Android phone. I was daft on one point, however, playlist and other features are available by swiping left, dumb of me. However even there previous, next and pause are not available.

**Reply by kan-ibal** [reply]
Then there is something wrong with your android. Check settings or system updates. I'm not familiar with android enough to give more hints.

**Reply by R101G** [reply]
Have you been through the Playback settings and set the top/bottom bars to show? What happens if you tap the top-centre of the cover art?

**Reply by debonaire007** [reply]
Thanks for helping guys, appreciated. Indeed, when you tap the top center of the covert art the controls appears :). There is also a very thin blue progress bar at the top edge of the covert art. This bar is functional and allows for moving  where in the song one wants to be. This feature is very subtle, not obvious on first sight, but cool. 

   On a another point, connecting to two or more rAudio systems does not seem to work.Under 'Features' when Multiple rAudios is selected , the pop up window does not allow any entries, it simply displays what the app is currently connected to Name and ip address.

**Reply by R101G** [reply]
So you have been through the settings menu and got the top/bottom bars showing?
I see you have the same problem I do with the oversized coverart, as you have accidentally set the random play mode when you meant to touch the Android back button :)

**Reply by debonaire007** [reply]
No the top bars are not showing, what works is tapping on the album cover at the top middle, this displays the controls.  To answer your point on 'oversized cover art', I do not think so, as random play is not selected. It may be that the cover art is somehow too large and the controls are hidden ... Overall however I like the app. 
   
What of my question of Multiple r Audio?

**Reply by R101G** [reply]
If the options are set to Bars Always On, I would expect them to show. The screenshot shows that you did have shuffle (random) enabled, and just picking up the phone can easily invoke the functions near the edge of the oversized coverart.

If you click on the question mark next to the Addons option in the menu, there is a cryptic user guide which might help you.

I have no idea what the multiple rAudio option is for.

**Comment by debonaire007** [comment]
Super thanks R101G. I add to my comment above that people on this Forum are knowlegeable and helpful By the way the Rune Audio app does not seem to find my RAudio devices, I have two of them working A o.k. :)
